cpp/include/raft/spatial/knn/detail/ann_kmeans_balanced.cuh

```diff
@@ -730,7 +832,7 @@ auto build_fine_clusters(const handle_t& handle,
  * @param metric the distance type
  * @param stream
  */
-template <typename T>
+template <typename T, typename IdxT, typename LabelT = int>
```

Comment:
Would you consider removing the default value for `LabelT` and move it to the first parameter? Since this is the only one that cannot be inferred. Gladly, it could only silently break the code in my draft PR at the moment, but still :)

Reply:
Yes, I could do that.
But my thinking was that it's actually better if the caller doesn't change that, at least for now. I could also have a `using` statement rather than a template parameter.

Reply:
That would make sense, but it's quite dangerous to let the type inference depend on a scalar parameter. It's very easy to overlook those types when calling the function and end up with 64-bit labels.

By the way, regarding unsigned label types, it seems that I only have to replace `-1` with `0` and it should work, in which case I can switch `LabelT` back to `uint32_t`. I'll try this out when I'm done debugging some overflows

Reply:
It's worth a try. Probably not a substantial difference for `fusedL2NN` if `dim` is large, but let's see.